includes/Core/Consent_Mode/REST_Consent_Mode_Controller.php

```diff
@@ -188,36 +188,35 @@ protected function get_rest_routes() {
 					array(
 						'methods'             => WP_REST_Server::READABLE,
 						'callback'            => function () {
-							$is_active  = function_exists( 'wp_set_consent' );
-							$installed  = $is_active;
-							$plugin_uri = 'https://wordpress.org/plugins/wp-consent-api';
-							$plugin     = 'wp-consent-api/wp-consent-api.php';
-
-							$response = array(
+							// Here we intentionally use a non-plugin-specific detection strategy.
+							$is_active = function_exists( 'wp_set_consent' );
+							$response  = array(
 								'hasConsentAPI' => $is_active,
 							);
 
+							// Alternate wp_nonce_url without esc_html breaking query parameters.
+							$nonce_url = function ( $action_url, $action ) {
+								return add_query_arg( '_wpnonce', wp_create_nonce( $action ), $action_url );
+							};
+
 							if ( ! $is_active ) {
-								$consent_mode_plugin_status = Plugin_Status::get_plugin_status( $plugin, $plugin_uri );
+								$installed_plugin = $this->get_consent_api_plugin_file();
```

Comment:
This actually fixes a bug that was present before since the `plugin` referenced in the `activateURL` below was always referencing `$plugin = 'wp-consent-api/wp-consent-api.php'`.